docs/src/man/basics.md

```diff
@@ -0,0 +1,2170 @@
+ "male"
+ "male"
+ "male"
+
+julia> german[!, :Sex] == german[:, :Sex]
+true
+```
+
+As it was explained above the difference between using `!` and `:`
+when passing a row index is that `!` does not perform a copy of columns, while `:` does.
+Therefore `german[!, [:Sex]]` data frame stores the same vector as the source `german` data frame,
+while `german[:, [:Sex]]` stores its copy. The `!` selector normally should be avoided
+as using it can lead to hard to catch bugs. However, when working with very large data frames
+it can be useful to save memory and improve performance of operations.
+
+
+Recapping what we have already learned,
+given the data frame `german` here are various ways to grab one of its columns as vector:
+
+```jldoctest dataframe
+julia> german
+1000×10 DataFrame
+  Row │ id     Age    Sex     Job    Housing  Saving accounts  Checking accoun ⋯
+      │ Int64  Int64  String  Int64  String   String           String          ⋯
+──────┼─────────────────────────────────────────────────────────────────────────
+    1 │     0     67  male        2  own      NA               little          ⋯
+    2 │     1     22  female      2  own      little           moderate
+    3 │     2     49  male        1  own      little           NA
+    4 │     3     45  male        2  free     little           little
+    5 │     4     53  male        2  free     little           little          ⋯
+    6 │     5     35  male        1  free     NA               NA
+    7 │     6     53  male        2  own      quite rich       NA
+    8 │     7     35  male        3  rent     little           moderate
+  ⋮   │   ⋮      ⋮      ⋮       ⋮       ⋮            ⋮                ⋮        ⋱
+  994 │   993     30  male        3  own      little           little          ⋯
+  995 │   994     50  male        2  own      NA               NA
+  996 │   995     31  female      1  own      little           NA
+  997 │   996     40  male        3  own      little           little
+  998 │   997     38  male        2  own      little           NA              ⋯
+  999 │   998     23  male        2  free     little           little
+ 1000 │   999     27  male        2  own      moderate         moderate
+                                                  4 columns and 985 rows omitted
+```
+
+We will get the vector stored in our DataFrame without copying it:
+
+```jldoctest dataframe
+julia> german.Age 
+1000-element Vector{Int64}:
+ 67
+ 22
+ 49
+ 45
+ 53
+ 35
+ 53
+ 35
+ 61
+ 28
+  ⋮
+ 34
+ 23
+ 30
+ 50
+ 31
+ 40
+ 38
+ 23
+ 27
+```
+
+We will get the vector stored in our data frame without copying it using String:
+
+```jldoctest dataframe
+julia> german."Sex" 
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+
+julia> german[:, 3] # note that this creates a copy
```

Comment:
`3` is not a string and the caption above signaled that you want to show string selection.